system-addon/lib/ActivityStreamMessageChannel.jsm

```diff
@@ -220,6 +232,15 @@ this.ActivityStreamMessageChannel = class ActivityStreamMessageChannel {
    * @param  {obj} msg The messsage from a page that was just loaded
    */
   onNewTabLoad(msg) {
+    let {browser} = msg.target;
+    if (this.isPreloadedBrowser(browser)) {
```

Comment:
Do you need to do this across windows as well? The `getPreloadedBrowser` function returns an array of all the preloaded browsers across all windows, and you can iterate over them to add `renderLayers` to each

Reply:
I don't think so, no - I think I only ever want to do this on a preloaded tab that just loaded. This is less about the _contents_ of the preloaded tab, and more about the preloadability of that one preloaded tab, if that makes sense. Once we receive the message that it has loaded, I'm making the parent say, "oh, okay, so we can paint and upload layers for this one tab then, since it just loaded", which increases the likelihood that when we do want to switch to it, it'll be more or less immediate.

Reply:
Ah makes sense. Thanks!